src/dotnet/commands/dotnet-compile/NativeCompiler.cs

```diff
@@ -22,16 +23,37 @@ public override bool Compile(ProjectContext context, CompilerCommandApp args)
 
             var managedOutput = outputPaths.CompilationFiles.Assembly;
 
+            // Create the library exporter
+            var exporter = context.CreateExporter(args.ConfigValue);
```

Comment:
The missing piece is the look up of `aot-win7-x64` (for example) assemblies and wiring it through to ILC skipping the ones that are in this list too.